### ISSUE TITLE
UI5 Linter fails to determine context of a call expression in common scenarios

### ISSUE THREAD

**Issue by flovogt** [UI5/linter#246]
When checking a call expression for deprecated methods, the linter [tries to understand the context of the call](https://github.com/SAP/ui5-linter/blob/main/src/linter/ui5Types/SourceFileLinter.ts#L174-L179) to later be able to include the context into its report. In certain common cases it fails to do so and throws an error like
`Unhandled CallExpression expression syntax: ${callee.type}`

Most prominent example (occurs quite often in modern ES2022 or TypeScript code when the constructor of a class calls `super(...)`):
```js
constructor() {
    super(id, settings);
}
```
results in `Unhandled CallExpression expression syntax: SuperKeyword` (found multiple times in sap.fe, e.g. in [@sapui5/sap.fe.templates/src/sap/fe/templates/ListReport/controls/MultipleModeControl.ts#76](https://www.npmjs.com/package/@sapui5/sap.fe.templates?activeTab=code)

Less prominent: when the call occurs on the result of another call:
```js
return newIsA.bind(controllerInstance)(className) || this.isA(className);
```
results in `Unhandled CallExpression expression syntax: CallExpression` (found in [@sapui5/sap.fe.core/src/sap/fe/core/ExtensionAPI.ts#297](https://www.npmjs.com/package/@sapui5/sap.fe.core?activeTab=code)).

## Context

- UI5 linter version: **0.2.2**
- Node.js Version: **v20.11.1**
- npm Version: 10.2.4
- OS/Platform: macOS 14.4.1

**Comment by flovogt** [comment]
Initially tracked SAP-internally via issue id 107

**Comment by matz3** [comment]
The first issue regarding `SuperKeyword` has already been fixed in `v0.2.3` via https://github.com/SAP/ui5-linter/pull/73.

The second issue will be fixed via https://github.com/SAP/ui5-linter/pull/255 (incl. a dedicated test case for the first issue).
